harness/determined/pytorch/_pytorch_context.py

```diff
@@ -800,6 +816,27 @@ def current_train_batch(self) -> int:
             raise det.errors.InternalException("Training hasn't started.")
         return self._current_batch_idx
 
+    @contextlib.contextmanager
+    def get_tensorboard_writer(self) -> Generator[tensorboard.MetricWriter, None, None]:
```

Comment:
This API is going to result in a _lot_ more reset() calls, right?

Why can't we call `.reset()` ourselves at the right times?  PyTorchTrial already sets tensorboard mode to "manual" I think (meaning PyTorchTrial already has careful control over the tensorboard behavior during training)

Reply:
I'm not exactly sure what a "right time" is here. Perhaps we can use the context method `self.is_epoch_end` ?